src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -122,10 +179,42 @@ async Task AuthenticateMD5(string username, byte[] salt, bool async)
             if (passwd == null)
                 throw new NpgsqlException("No password has been provided but the backend requires one (in MD5)");
 
-            await PasswordMessage
-                .CreateMD5(passwd, username, salt)
-                .Write(WriteBuffer, async);
-            await WriteBuffer.Flush(async);
+            var md5 = MD5.Create();
+
+            // First phase
+            var passwordBytes = PGUtil.UTF8Encoding.GetBytes(passwd);
+            var usernameBytes = PGUtil.UTF8Encoding.GetBytes(username);
```

Comment:
Do we still need our version, or can we use the built-in?

Reply:
You mean whether we should stop using `PGUtil.UTF8Encoding` in general? That's just a cached version of `UTF8Encoding` configured the way we need it (e.g. throw on invalid bytes). If you think there's something to be improved, let's take a look at that separately as it affects the entire codebase.

Reply:
Oh, in that case, nevermind! Assumed this was around to support older framework versions.